### PR TITLE
Modified `onUpdate` method

### PR DESCRIPTION
**Pull request by afzal442**
### Type of change:
Added type assertions to check that the old and new objects are of type *gatewayv1beta1.Gateway. Then, checked if the resource versions of the old and new objects are the same. If they are, it means the update is for an outdated version of the object, so ignore it. Otherwise, get the key for the new object and add it to the work queue.

- [ ] Bugfix
- [ ] New feature provided
- [ ] Improve performance
- [ ] Backport patches
- [ ] Documentation
- [x] Refactor
- [ ] Chore
- [ ] CI/CD or Tests

### Pre-submission checklist:

- [ ] Did you explain what problem does this PR solve? Or what new features have been added?
- [ ] Have you added corresponding test cases?
- [ ] Have you modified the corresponding document?
- [ ] Is this PR backward compatible? **If it is not backward compatible, please discuss on the [mailing list](https://github.com/apache/apisix-ingress-controller#community) first**

### CODE REVIEW

**Comment by tao12345666333** [comment]
Thanks for your contributions.

Please add an e2e test case for it.

**Comment by afzal442** [comment]
Sure! I will try to do that.

File: pkg/providers/gateway/gateway.go [hunk 1]
```diff
@@ -206,6 +207,49 @@ func (c *gatewayController) onAdd(obj interface{}) {
...
+
+	newGateway, ok := newObj.(*gatewayv1beta1.Gateway)
+	if !ok {
+		log.Errorw("failed to convert new object to gateway",
+			zap.Any("new obj", newObj),
+		)
+		return
+	}
+
+	if reflect.DeepEqual(oldGateway.Spec, newGateway.Spec) {
```
Comment (afzal442) [inline pkg/providers/gateway/gateway.go:226]:
```suggestion
	if oldHTTPRoute.ResourceVersion >= newHTTPRoute.ResourceVersion {
```
I just wanted to know if we can change it to this one or previous one is okay.

**Comment by afzal442** [comment]
Hi! looks like there are already test cases https://github.com/apache/apisix-ingress-controller/blob/2182a48cbca785373eca745a13d8cf2b7d9ab6c8/test/e2e/suite-features/external-sd.go#L235
And I think we have test failed due to this https://github.com/apache/apisix-ingress-controller/blob/2182a48cbca785373eca745a13d8cf2b7d9ab6c8/test/e2e/suite-features/external-service.go#L289

I will try again with `make e2e-test-local`. Any thoughts?

**Comment by afzal442** [comment]
hi @tao12345666333 as a friendly reminder, can I have some review for this? ⛏️ Are the tests faliing related to it? 
One more question, is there e2e-test written for OnAdd() and OnDelete() methods of Gateway controller? Thanks

**Comment by tao12345666333** [comment]
Not yet. 

All tests related to Gateway API are stored in this directory: https://github.com/apache/apisix-ingress-controller/tree/master/test/e2e/suite-gateway

**Comment by afzal442** [comment]
Hey @tao12345666333 ! do you think I will have to write e2e test case for that as there is no?

**Comment by tao12345666333** [comment]
yes, please!

**Pull request closed without merging**